lib/purescript-ast/src/Language/PureScript/AST/Declarations.hs

```diff
@@ -429,7 +431,7 @@ data Declaration
   -- A type instance declaration (instance chain, chain index, name,
   -- dependencies, class name, instance types, member declarations)
   --
-  | TypeInstanceDeclaration SourceAnn [Ident] Integer Ident [SourceConstraint] (Qualified (ProperName 'ClassName)) [SourceType] TypeInstanceBody
+  | TypeInstanceDeclaration SourceAnn [ChainId] Integer (Either Text Ident) [SourceConstraint] (Qualified (ProperName 'ClassName)) [SourceType] TypeInstanceBody
```

Comment:
Can this be `ChainId` instead of `[ChainId]` (here and everywhere)? Identifying each chain by the position of its first instance should work just as well as identifying them by the list of the positions of all their instances, right?